src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -41,6 +45,26 @@
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        StringBuilder sb = new StringBuilder("Отчёт за период " + dateFrom + " - " + dateTo + "\n");
+        SimpleDateFormat format = new SimpleDateFormat();
+        format.applyPattern("dd.MM.yyyy");
+        Date dateFromDate = format.parse(dateFrom);
+        Date dateToDate = format.parse(dateTo);
+        if (dateFromDate.compareTo(dateToDate) == 1) {
```

Comment:
try to solve without `compareTo` at all. there are more relevant methods

Reply:
done